components/security_reporter/sources/common/base.mjs

```diff
@@ -0,0 +1,48 @@
+import { randomUUID } from "crypto";
+import securityReporter from "../../security_reporter.app.mjs";
+
+export default {
+  props: {
+    securityReporter,
+    http: {
+      type: "$.interface.http",
+      customResponse: false,
+    },
+    db: "$.service.db",
+    name: {
+      propDefinition: [
+        securityReporter,
+        "name",
+      ],
+    },
+  },
+  hooks: {
+    async activate() {
+      const { id } = await this.securityReporter.createWebhook({
+        data: {
+          url: this.http.endpoint,
+          name: this.name,
+          secret: randomUUID(),
+          auth_method: 0,
+          types: this.getTypes(),
+          mode: 0,
+        },
+      });
+      this.db.set("webhookId", id);
+    },
+    async deactivate() {
+      const webhookId = this.db.get("webhookId");
+      if (webhookId) {
+        await this.securityReporter.deleteWebhook(webhookId);
+      }
+    },
+  },
+  async run({ body }) {
+    const ts = Date.parse(body.model.updated_at);
+    this.$emit(body, {
+      id: `${body.model.id}-${ts}`,
+      summary: this.getSummary(body.model),
+      ts,
+    });
```

Comment:
According to the test events, `body.model` doesn't exist for `new-assessment-instant`. I would create a `generateMeta` method that can be overwritten by the `new-assessment-instant` source. I'm not sure which fields would work best for `id` and `summary`, but you could use `Date.now()` for the `ts`.